contracts/btc-finality/src/contract.rs

```diff
@@ -240,26 +244,26 @@ fn handle_end_block(
     _hash_hex: &str,
     app_hash_hex: &str,
 ) -> Result<Response, ContractError> {
-    // If the BTC staking protocol is activated i.e. there exists a height where at least one
-    // finality provider has voting power, start indexing and tallying blocks
-    let cfg = CONFIG.load(deps.storage)?;
     let mut res = Response::new();
+    let cfg = CONFIG.load(deps.storage)?;
 
-    let activated_height = get_btc_staking_activated_height(deps.storage);
+    // Should not proceed if not reached finality activation height
+    if env.block.height < cfg.finality_activation_height {
+        return Ok(res);
+    }
 
-    // If the BTC staking protocol is not activated, do nothing
+    // Should not proceed if BTC staking is not activated
+    let activated_height = get_btc_staking_activated_height(deps.storage);
     if activated_height.is_none() {
         return Ok(res);
     }
 
-    let activated_height = activated_height.unwrap();
-
     // Index the current block
     let ev = finality::index_block(deps, env.block.height, &hex::decode(app_hash_hex)?)?;
     res = res.add_event(ev);
 
     // Tally all non-finalised blocks
-    let events = finality::tally_blocks(deps, &env, activated_height)?;
+    let events = finality::tally_blocks(deps, &env, activated_height.unwrap())?;
     res = res.add_events(events);
```

Comment:
Here tallying should start since `max(activated_height, cfg.finality_activation_height)`

Reply:
There are still a few mislignments in tally_blocks but out of the scope of this pr. Tracked in https://github.com/babylonlabs-io/cosmos-bsn-contracts/issues/306. Low priority though

Reply:
Also the parameter name in `tally_blocks` could probably be renamed. Currently it's called `activated_height` which might not be precise anymore. Minor stuff though

On the misalignment, I think the two versions are equivalent, and the version in contract is more clean imo actually

Reply:
yeah, this is because we don't want to load activated_height again but passing it as input of `tally_blocks`. We will figure out something when addressing #306